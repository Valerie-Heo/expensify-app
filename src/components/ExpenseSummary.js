import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import SelectExpenses from "../selectors/expenses";
import SelectExpensesTotal from "../selectors/expenseTotal";

export const ExpenseSummary = ({ expenseCount, expenseCost }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const FormatedexpenseCost = numeral(expenseCost / 100).format("0, 000.00");
  return (
    <div>
      <h1>  Viewing {expenseCount} {expenseWord} with at total cost that equal {FormatedexpenseCost} </h1>
    </div>
  );
};

const mapToStateProp = (state) => {
  const visibleExpenses = SelectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expenseCost: SelectExpensesTotal(visibleExpenses),
  };
};

export default connect(mapToStateProp)(ExpenseSummary);
