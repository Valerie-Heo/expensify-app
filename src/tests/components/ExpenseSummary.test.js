import { shallow } from "enzyme";
import React from "react";
import { ExpenseSummary } from "../../components/ExpenseSummary";

test("this should render Expense Summary without any expenses ", () => {
  const wrapper = shallow(
    <ExpenseSummary expenseCount={1} expenseCost={195} />
  );
  expect(wrapper).toMatchSnapshot();
});


test("this should render Expense Summary without any expenses ", () => {
  const wrapper = shallow(
    <ExpenseSummary expenseCount={13} expenseCost={223195} />
  );
  expect(wrapper).toMatchSnapshot();
});
