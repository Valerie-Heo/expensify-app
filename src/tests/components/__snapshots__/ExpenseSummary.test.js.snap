// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`this should render Expense Summary without any expenses  1`] = `
<div>
  <h1>
     Viewing 
    1
     
    expense
     with at total cost that equal 
    1.95
     
  </h1>
</div>
`;

exports[`this should render Expense Summary without any expenses  2`] = `
<div>
  <h1>
     Viewing 
    13
     
    expenses
     with at total cost that equal 
    2,231.95
     
  </h1>
</div>
`;
